test/unit/messaging/messaging.spec.ts

```diff
@@ -566,6 +566,12 @@ describe('Messaging', () => {
         { token: 'mock-token' },
       ).should.eventually.be.rejected.and.have.property('code', 'messaging/invalid-argument');
     });
+
+    it('should not throws when the payload contains the data property with undefined value', () => {
```

Comment:
This test case is actually irrelevant to the actual fix. The fix only applies to legacy FCM APIs like `sendToDevice()`. It has no effect on `send()`. We can remove this.